build-system/common/git.js

```diff
@@ -27,32 +27,32 @@ const {
 } = require('./travis');
 const {getStdout} = require('./exec');
 
+/**
+ * Returns the remote tracking branch associated with the local master branch.
+ * @return {string}
+ */
+function gitUpstreamMaster() {
+  const upstreamName = getStdout('git config branch.master.remote').trim();
+  return (upstreamName || 'origin') + '/master';
+}
+
 /**
  * Returns the commit at which the current branch was forked off of master.
  * On Travis, there is an additional merge commit, so we must pick the first of
  * the boundary commits (prefixed with a -) returned by git rev-list.
- * On local branches, this is merge base of the current branch off of master.
+ * On local branches, this is merge base of the current branch off of
+ * upstream/master.
  * @return {string}
  */
 function gitBranchCreationPoint() {
   if (isTravisBuild()) {
```

Comment:
I wonder if this whole block can be deleted.

Reply:
Unless, I'm misunderstanding how Travis runs, I believe
```
git rev-list --boundary ${traviPrSha}...master | grep "^-" | head -n 1 | cut -c2-
```
produces the same hash as
```
git merge-base origin/master HEAD
```
assuming the active branch is the one in which `traviPrSha` was merged.

Reply:
While `git merge-base` works for simple cases, it breaks down for PR branches with multiple merge commits. (E.g. A developer refreshes a PR branch multiple times via merge commits instead of rebases.)

See #22355 for more context about why we had to switch to `git rev-list`. Therefore, I'm pretty sure this part of your PR will have to be reverted.

Reply:
Thanks I was able to reproduce the issue with multiple merges. PR significantly toned down from version you originally reviewed.

Reply:
Marking this comment as resolved since updated PR does not touch Travis code paths anymore.